enforcer/datapath/tokenaccessor/tokenaccessor.go

```diff
@@ -86,37 +86,39 @@ func (t *tokenAccessor) CreateAckPacketToken(context *pucontext.PUContext, auth
 // createSynPacketToken creates the authentication token
 func (t *tokenAccessor) CreateSynPacketToken(context *pucontext.PUContext, auth *connection.AuthInfo) (token []byte, err error) {
 
-	if context.SynExpiration.After(time.Now()) && len(context.SynToken) > 0 {
+	token, err = context.GetCachedToken()
+	if err == nil {
```

Comment:
Can you change this to do the opposite?

Usually if err == nil should be used only in some specific cases. Always prefer if err != nil

Reply:
I agree. In this case,  if else constructs are kept to a minimum